### DISCUSSION TITLE
User-created function to produce formulas with integral

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#3146] [Q&A] [answered]
I am trying to automate the creation of a certain type of formulas, by creating a function like the following:

```
#let intpart(G, F) = {
  integral underbracket(#G, G) underbracket(#F, F) dif x
}
```

By calling:

```
$
#intpart(x^2, cos(x))
$
```

the result should be the following:

![image](https://github.com/typst/typst/assets/75945439/93be09d3-9ac5-4e63-9b02-2ba4c5c222e7)

Unfortunately, I get errors. Could you please help this `typst` beginner?

**Comment by frozolotl** [comment, marked as answer]
You have to wrap the math code in the function inside `$…$`.
When calling it, you must not use `#`, else Typst switches to code mode and math syntax like `x^2` is no longer allowed.

```typ
#let intpart(G, F) = {
  $integral underbracket(#G, G) underbracket(#F, F) dif x$
}

$ intpart(x^2, cos(x)) $
```

**Reply by psads-git** [reply]
Perfect! Thanks, @frozolotl!
